File: src/main/java/bad/robot/radiate/monitor/Observable.java
package bad.robot.radiate.monitor;

import bad.robot.radiate.Activity;
import bad.robot.radiate.Progress;
import bad.robot.radiate.Status;

import java.util.stream.Stream;

@Deprecated
public interface Observable {
    void addObservers(Stream<Observer> observer);
    boolean addObservers(Observer... observer);
    boolean removeObservers(Observer... observer);
    void removeAllObservers();
    void notifyObservers(Status status);
    void notifyObservers(Activity activity, Progress progress);
    void notifyObservers(Exception exception);
    void notifyObservers(Information information);
}

File: src/main/scala/bad/robot/radiate/monitor/Observable.scala
package bad.robot.radiate.monitor

import bad.robot.radiate.RadiateError.Error
import bad.robot.radiate._

trait Observable {
  def addObservers(observer: List[Observer])
  def addObservers(observer: Observer*): Boolean
  def removeObservers(observer: Observer*): Boolean
  def removeAllObservers()
  def notifyObservers(status: Status)
  def notifyObservers(activity: Activity, progress: Progress)
  def notifyObservers(error: Error)
  def notifyObservers(exception: Exception)
  def notifyObservers(information: Information)
}
